src/main/java/org/jsoup/helper/W3CDom.java

```diff
@@ -96,6 +97,35 @@ public static Document convert(org.jsoup.nodes.Document in) {
         return (new W3CDom().fromJsoup(in));
     }
 
+    /**
+     * Pattern to detect the <code>xmlns="http://www.w3.org/1999/xhtml"</code> default namespace
+     * declaration when serializing the DOM to HTML. This pattern is "good enough", relying in part
+     * on the output of the {@link Transformer} used in the implementation, but is not a complete
+     * solution for all the serializations possible; that is, if one constructed an XML string
+     * manually, it might be possible to find an obscure variation that this pattern would not
+     * match.
+     */
+    static final Pattern HTML_DEFAULT_NAMESPACE_PATTERN =
+        Pattern.compile("<html[^>]*(\\sxmlns=['\"]http://www.w3.org/1999/xhtml['\"])");
+
+    /**
+     * Removes the default <code>xmlns="http://www.w3.org/1999/xhtml"</code> HTML namespace
+     * declaration if present in the string.
+     * 
+     * @param html The serialized HTML.
+     * @return A string without the default <code>xmlns="http://www.w3.org/1999/xhtml"</code> HTML
+     *         namespace declaration.
+     * @see <a href="https://github.com/jhy/jsoup/issues/1837">Issue #1837: Bug: DOM elements not
+     *      being placed in (X)HTML namespace.</a>
+     */
+    static String removeDefaultHtmlNamespaceDeclaration(String html) {
+        Matcher matcher = HTML_DEFAULT_NAMESPACE_PATTERN.matcher(html);
+        if (matcher.find()) {
+          html = html.substring(0, matcher.start(1)) + html.substring(matcher.end(1));
```

Comment:
I think it is better to leave the introduced namespace in the output. It makes the output XML clearer as to what's happened, and may help other downstream parses. So the regex components etc can be removed.

> We can leave it like that or play tricks to get rid of it, but it depends on the purpose and usage of W3CDom.asString(). Is it just used for testing? Or is it part of the jsoup API for turning DOM into a string?

It's part of the API.

> jsoup is about parsing, not pretty-printing

I consider jsoup to be also about pretty-printing. And we do introduce other tokens throughout jsoup as a convenience (e.g. xml declarations, meta charsets, the HTML document structure, etc).

Reply:
> I think it is better to leave the introduced namespace in the output.

For my purposes I don't care one way or another, and it's certainly easier just to leave it in. At the time, due to lack of response to the ticket, I had to make a choice and do what I thought would raise the chance of the PR being requested. (You can see all the time and painful [research on Stack Overflow](https://stackoverflow.com/q/73919306) to remove the declaration.) I'll remove the code that removed the declaration in the output.